stream-portfolio/portfolio-core/src/main/java/com/backbase/stream/portfolio/PortfolioSaga.java

```diff
@@ -0,0 +1,128 @@
+package com.backbase.stream.portfolio;
+
+import com.backbase.stream.config.BackbaseStreamConfigurationProperties;
+import com.backbase.stream.portfolio.exceptions.PortfolioBundleException;
+import com.backbase.stream.portfolio.service.InstrumentIntegrationService;
+import com.backbase.stream.portfolio.service.PortfolioIntegrationService;
+import com.backbase.stream.portfolio.service.ReactiveStreamHandler;
+import com.backbase.stream.webclient.DbsWebClientConfiguration;
+import com.backbase.stream.worker.StreamTaskExecutor;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.sleuth.annotation.ContinueSpan;
+import org.springframework.cloud.sleuth.annotation.SpanTag;
+import org.springframework.context.annotation.Import;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+@EnableConfigurationProperties(BackbaseStreamConfigurationProperties.class)
+@Import(DbsWebClientConfiguration.class)
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class PortfolioSaga implements StreamTaskExecutor<PortfolioTask> {
```

Comment:
in this saga, although all the methods are called `upsert` I don't see the logic to validate if they are existing before creating here. Is this implementation idempotent?

Reply:
yeah, correct observation, change the name to `insert`. suppose during support check for exist will be added